OTAnalytics/plugin_ui/nicegui_gui/pages/visualization_filters_form/container.py

```diff
@@ -0,0 +1,219 @@
+                    ui.checkbox(on_change=lambda e: self.change_button_date(e.value))
+                    self._filter_by_date_button = ui.button(
+                        self._resource_manager.get(
+                            VisualizationFiltersKeys.BUTTON_FILTER_BY_DATE
+                        ),
+                        on_click=self.open_range_dialog,
+                    )
+                    self._filter_by_date_button.disable()
+
+                ui.button("<", on_click=self._viewmodel.previous_second)
+                ui.button("<", on_click=self._viewmodel.previous_frame)
+                ui.button("<", on_click=self._viewmodel.previous_event)
+            with ui.column():
+                with ui.row():
+                    self.filter_by_date_button_left = ui.button(
+                        "<", on_click=self._viewmodel.switch_to_prev_date_range
+                    )
+                    self.label_filter_by_date = ui.label()
+                    self.filter_by_date_button_right = ui.button(
+                        ">", on_click=self._viewmodel.switch_to_next_date_range
+                    )
+                    self.filter_by_date_button_left.disable()
+                    self.filter_by_date_button_right.disable()
+                self.input_seconds = ui.number(
+                    value=self._viewmodel.get_skip_seconds(),
+                    on_change=self.on_input_change,
+                )
+                self.input_frames = ui.number(
+                    value=self._viewmodel.get_skip_frames(),
+                    on_change=self.on_input_change,
+                )
+                ui.label(
+                    self._resource_manager.get(VisualizationFiltersKeys.LABEL_EVENT)
+                )
+            with ui.column():
+                with ui.row():
+                    self._filter_by_class_checkbox = ui.checkbox(
+                        on_change=lambda e: self.change_button_class(e.value)
+                    )
+                    self._filter_by_class_button = ui.button(
+                        self._resource_manager.get(
+                            VisualizationFiltersKeys.BUTTON_FILTER_BY_CLASSIFICATION
+                        ),
+                        on_click=self.open_classification_dialog,
+                    )
+                    self._filter_by_class_button.disable()
+                ui.button(">", on_click=self._viewmodel.next_second)
+                ui.button(">", on_click=self._viewmodel.next_frame)
+                ui.button(">", on_click=self._viewmodel.next_event)
+
+    def change_button_date(self, checkbox_status: bool) -> None:
+        self._filter_by_date_button.set_enabled(checkbox_status)
+        self.filter_by_date_button_left.set_enabled(checkbox_status)
+        self.filter_by_date_button_right.set_enabled(checkbox_status)
+
+    def change_button_class(self, checkbox_status: bool) -> None:
+        self._filter_by_class_button.set_enabled(checkbox_status)
```

Comment:
This logic is covered by the viewmodel. Use self._viewmodel.enable_filter_track_by_class and disable_filter_track_by_class. See more in FilterTracksByClassFilterButton#_enable_filter